napari/components/_interaction_box_mouse_bindings.py

```diff
@@ -77,9 +80,16 @@ def _on_add_layer(self, event):
         if hasattr(layer, 'mode'):
             layer.events.mode.connect(self._on_mode_change)
 
-    def _on_active(self, event):
+    def _on_active(self, event=None):
         """Gets called when active layer is changed"""
-        active_layer = event.value
+        viewer = self._ref_viewer()
+        active_layer = viewer.layers.selection.active
+        if active_layer is not None:
+            active_layer.events.rotate.connect(self._on_active)
+            active_layer.events.translate.connect(self._on_active)
+            active_layer.events.scale.connect(self._on_active)
+            active_layer.events.shear.connect(self._on_active)
```

Comment:
general question not specific to this PR, I see these events get connected but not disconnected, is this handled anywhere?